azkaban-web-server/src/main/java/cloudflow/services/ExecutionServiceImpl.java

```diff
@@ -1,126 +1,201 @@
-                }
-            }
-            if (i >= nodesToScan.size()) {
-                errorMsg = String.format("Job with id %s and path %s wasn't found in execution %d.",
-                    jobDefinitionId, jobPath, execId);
-                logger.error(errorMsg);
-                throw new CloudFlowNotFoundException(errorMsg);
-            }
-            if (node instanceof ExecutableFlowBase) {
-                nodesToScan = ((ExecutableFlowBase) node).getExecutableNodes();
-            } else {
-                // If we find a job before the end of the path this ensures the next element in
-                // the path will not be found in the next iteration. This has no effect with the
-                // last node in the path.
-                nodesToScan = new ArrayList<>();
-            }
-        }
-        logger.info("Found job with path {} in execution {}.", jobPath, execId);
+public class ExecutionServiceImpl implements ExecutionService {
 
-        JobExecution jobExecution = new JobExecution();
-        List<JobExecutionAttempt> attempts = getJobExecutionAttempts(node);
-        jobExecution.setAttempts(attempts);
+  private static final String FLOW_ID_PARAM = "flow_id";
+  private static final String FLOW_VERSION_PARAM = "flow_version";
+  private static final String PROJECT_ID_PARAM = "project_id";
+  private static final String EXPERIMENT_ID_PARAM = "experiment_id";
+
+  private static final Logger logger = LoggerFactory.getLogger(ExecutionServiceImpl.class);
+  private ExecutorManagerAdapter executorManager;
+  private ExecutionFlowDao executionFlowDao;
+
+  @Inject
+  public ExecutionServiceImpl(ExecutorManagerAdapter executorManager,
+      ExecutionFlowDao executionFlowDao) {
+    this.executorManager = executorManager;
+    this.executionFlowDao = executionFlowDao;
+  }
+
+  private String extractArgumentValue(String argName, String[] argValues) {
+    requireNonNull(argName);
+    requireNonNull(argValues);
+    if (argValues.length != 1) {
+      throw new RuntimeException(format("Argument %s should have exactly one value", argName));
+    }
+    return argValues[0];
+  }
+
+  private ExecutionBasicResponse executionResponseFromFlow(ExecutableFlow executableFlow) {
+    return new ExecutionBasicResponse(executableFlow);
+  }
+
+  @Override
+  public List<ExecutionBasicResponse> getAllExecutions(Map<String, String[]> queryParamMap) {
+    requireNonNull(queryParamMap, "query param map is null");
+
+    // Currently we support only one value for each of the parameters and return an error otherwise.
+    Optional<String> flowId = Optional.empty();
+    Optional<String> flowVersion = Optional.empty();
+    Optional<String> experimentId = Optional.empty();
+    Optional<String> projectId = Optional.empty();
+
+    for (Map.Entry<String, String[]> entry : queryParamMap.entrySet()) {
+      String key = entry.getKey();
+      String[] value = entry.getValue();
+      if (key.equalsIgnoreCase(FLOW_ID_PARAM)) {
```

Comment:
Wondering if it makes sense to make these _equals_ case sensitive to help us enforce naming conventions.

Reply:
Good point. Will be specially relevant in case we decide to use `camelCase` params. Will fix.